libcudacxx/include/cuda/std/__iterator/wrap_iter.h

```diff
@@ -59,67 +50,23 @@ class __wrap_iter
 public:
   _LIBCUDACXX_HIDE_FROM_ABI _CCCL_CONSTEXPR_CXX14 __wrap_iter() noexcept
       : __i_()
-  {
-    _CUDA_VSTD::__debug_db_insert_i(this);
-  }
+  {}
   template <class _Up>
   _LIBCUDACXX_HIDE_FROM_ABI _CCCL_CONSTEXPR_CXX14
   __wrap_iter(const __wrap_iter<_Up>& __u,
               typename enable_if<is_convertible<_Up, iterator_type>::value>::type* = nullptr) noexcept
       : __i_(__u.base())
-  {
-#ifdef _LIBCUDACXX_ENABLE_DEBUG_MODE
-    if (!__libcpp_is_constant_evaluated())
-    {
-      __get_db()->__iterator_copy(this, _CUDA_VSTD::addressof(__u));
-    }
-#endif
-  }
-#ifdef _LIBCUDACXX_ENABLE_DEBUG_MODE
-  _LIBCUDACXX_HIDE_FROM_ABI _CCCL_CONSTEXPR_CXX14 __wrap_iter(const __wrap_iter& __x)
-      : __i_(__x.base())
-  {
-    if (!__libcpp_is_constant_evaluated())
-    {
-      __get_db()->__iterator_copy(this, _CUDA_VSTD::addressof(__x));
-    }
-  }
-  _LIBCUDACXX_HIDE_FROM_ABI _CCCL_CONSTEXPR_CXX14 __wrap_iter& operator=(const __wrap_iter& __x)
-  {
-    if (this != _CUDA_VSTD::addressof(__x))
-    {
-      if (!__libcpp_is_constant_evaluated())
-      {
-        __get_db()->__iterator_copy(this, _CUDA_VSTD::addressof(__x));
-      }
-      __i_ = __x.__i_;
-    }
-    return *this;
-  }
-  _LIBCUDACXX_HIDE_FROM_ABI _CCCL_CONSTEXPR_CXX20 ~__wrap_iter()
-  {
-    if (!__libcpp_is_constant_evaluated())
-    {
-      __get_db()->__erase_i(this);
-    }
-  }
-#endif
```

Comment:
Question: Why remove all this? And why as part of this PR? It seems unrelated to the assert unification.

Reply:
Thats something i had in that branch lying around. We need to get rid of all the old debug handling